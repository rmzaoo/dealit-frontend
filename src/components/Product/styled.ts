import styled from "styled-components";

const Container = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 600px;
  height: 350px;
  color: black;
  border-radius: 10px;
  box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.75);
  background-color: #ffffff;
  background-image: linear-gradient(135deg, #ffffff 54%, #aeaeae 100%);

  > * {
    margin: 0px 10px;
  }
`;

const ProductPhotoDiv = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 450px;
  height: 100%;

  img {
    width: 100%;
    height: 100%;
    object-fit: scale-down;
  }
`;

const ProductDetails = styled.div`
  height: 70%;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-end;
  text-align: right;
  margin: 0px 20px;


  h1 {
    font-size: 30px;
    font-weight: bold;
  }

  span {
    font-size: 20px;
    letter-spacing: 1.5px;
  }
`;

export { Container, ProductPhotoDiv, ProductDetails };
