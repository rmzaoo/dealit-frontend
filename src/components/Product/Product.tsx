import React from "react";
import { useNavigate } from "react-router-dom";
import { PrimaryButton } from "../PrimaryButton/style";
import { Container, ProductDetails, ProductPhotoDiv } from "./styled";
import noImage from '../../assets/noImageAvailable.png';

export interface ProductProps {
  id: number;
  name: string;
  photos: string[];
  price: number;
  className?: string;
}

const Product = ({ id, name, photos, price, className }: ProductProps) => {
  const navigate = useNavigate();
  return (
    <Container className={className} onClick={() => navigate(`/product/${id}`)}>
      <ProductPhotoDiv>
        <img
          src={photos[0]}
          onError={(e) => (e.currentTarget.src = noImage)}
          alt="product"
        />
      </ProductPhotoDiv>
      <ProductDetails>
        <h1>{name}</h1>
        <span>Price: ${price}</span>
        {/* <section>
          <PrimaryButton>View Product</PrimaryButton>
        </section> */}
      </ProductDetails>
    </Container>
  );
};

export default Product;
