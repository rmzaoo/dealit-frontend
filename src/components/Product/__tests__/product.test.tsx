import { describe, test, expect, vi } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Product from "../Product";

describe("product", () => {
  test("should first", () => {
    const result = render(
      <BrowserRouter>
        <Product
          id={1}
          name="test"
          photos={["https://via.placeholder.com/300"]}
          price={1}
        />
      </BrowserRouter>
    );

    // const img = screen.getByAltText("product");
    // expect(img.getAttribute("src")).toBe("https://via.placeholder.com/300");

    // expect(screen.getByText("test"));
    // expect(screen.getByText("1"));

    // expect(result).toMatchSnapshot();
  });
});
