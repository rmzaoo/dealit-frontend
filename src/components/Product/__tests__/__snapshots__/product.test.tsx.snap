// Vitest Snapshot v1

exports[`product > renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ fnHYht"
      >
        <div
          class="sc-gsnTZi jbBFer"
        >
          <img
            alt="product"
            src="https://via.placeholder.com/300"
          />
        </div>
        <div
          class="sc-dkzDqf fLVvgG"
        >
          <h1>
            test
          </h1>
          <span>
            Price: $
            1
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ fnHYht"
    >
      <div
        class="sc-gsnTZi jbBFer"
      >
        <img
          alt="product"
          src="https://via.placeholder.com/300"
        />
      </div>
      <div
        class="sc-dkzDqf fLVvgG"
      >
        <h1>
          test
        </h1>
        <span>
          Price: $
          1
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`product > should first 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ hZLqxf"
      >
        <div
          class="sc-gsnTZi gmHuzv"
        >
          <img
            alt="product"
            src="https://via.placeholder.com/300"
          />
        </div>
        <div
          class="sc-dkzDqf fLVvgG"
        >
          <h1>
            test
          </h1>
          <span>
            Price: $
            1
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ hZLqxf"
    >
      <div
        class="sc-gsnTZi gmHuzv"
      >
        <img
          alt="product"
          src="https://via.placeholder.com/300"
        />
      </div>
      <div
        class="sc-dkzDqf fLVvgG"
      >
        <h1>
          test
        </h1>
        <span>
          Price: $
          1
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
